frontend/.claude/skills/frontend-developer/SKILL.md

```diff
@@ -0,0 +1,304 @@
+  const actual = await importOriginal<typeof import("module-name")>();
+  return {
+    ...actual,
+    functionToMock: vi.fn(),
+  };
+});
+
+const mockFunction = vi.mocked(functionToMock);
+```
+
+### 5. Validation
+
+```bash
+# Run in order
+bun run type:check    # TypeScript errors
+bun run test          # All tests
+bun run lint          # Code quality
+bun run build         # Production build
+bun run start         # Dev server - check browser
+```
+
+**Success criteria:**
+
+- ✓ No TypeScript errors
+- ✓ All tests passing
+- ✓ No lint errors
+- ✓ Build succeeds
+- ✓ No runtime errors in browser
+
+## Testing Commands
+
+```bash
+bun run test                 # All tests (unit + integration)
+bun run test:unit            # Unit tests only (.test.ts)
+bun run test:integration     # Integration tests only (.test.tsx)
+bun run test:watch           # Watch mode
+bun run test:ui              # Interactive UI
+bun run test:coverage        # Coverage report
+```
+
+## Common Patterns
+
+### Registry Component Usage
+
+Check `src/components/redpanda-ui/` for installed components before installing new ones.
+
+Use component variants instead of custom styling:
+
+```typescript
+<Button variant="primary" size="lg" />
+<Card variant="outline" />
+<Dialog size="md" />
+```
+
+### Form Patterns
+
+For complex forms, use React Hook Form + Zod:
+
+```typescript
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+
+const schema = z.object({
+  name: z.string().min(1),
+  email: z.string().email(),
+});
+
+const form = useForm({
+  resolver: zodResolver(schema),
+  defaultValues: { name: "", email: "" },
+});
+```
+
+For simple forms, use AutoForm:
+
+```typescript
+import { AutoForm } from '@autoform/react';
+import { ZodProvider } from '@autoform/zod';
+
+<AutoForm
+  schema={schema}
+  onSubmit={handleSubmit}
+  provider={ZodProvider}
+/>
+```
+
+### Reusable Sub-Components
+
+Create sub-components in the same file when logic repeats:
+
+```typescript
+export function FeatureComponent() {
+  return (
+    <div>
+      <FeatureHeader />
+      <FeatureContent />
+    </div>
+  );
+}
+
+// Colocated sub-components
+function FeatureHeader() { /* ... */ }
+function FeatureContent() { /* ... */ }
+```
+
+## Testing Best Practices
+
+**File naming:**
+
+- `.test.ts` for utilities and pure functions (unit tests)
+- `.test.tsx` for React components (integration tests)
+
+**Mock location:**
+
+- Use pre-configured mocks from `src/test-utils/`
+- Mock external dependencies, not internal code
+
+**Render utility:**
+
+- Import from `test-utils/test-utils.tsx` for component tests
+- Includes all providers (React Query, Router, Auth0)
+
+**Async testing:**
+
+- Use `waitFor` for async operations
+- Test loading and error states
+- Verify state updates
+
+## Quick Reference
+
+**Check before installing:**
+
+```bash
+# Check if component exists
+ls src/components/redpanda-ui/components/
```

Comment:
Maybe say if file exists and then `ls src/components/redpanda-ui/` because we may also have `hooks` or other directories.